edx/edx-enterprise
# [ENT-437] Audit listed above Verified

**Description:** This PR sorts the course modes of the landing page by slug based on the `enterprise.constants.COURSE_MODE_SORT_ORDER` list.

**JIRA:** [ENT-437](https://openedx.atlassian.net/browse/ENT-437)

**Screenshots:**
![screenshot](https://user-images.githubusercontent.com/560781/27349205-bf5119ca-55b3-11e7-9cca-1336cca82ab9.png)

**Dependencies:** None. 

**Merge deadline:**  04/Jul/17

**Installation instructions:**  Standard edx-enterprise installation.

**Testing instructions:**

1. Set up this branch in a devstack.
2. Start the ecommerce service.
3. In the LMS, register a new user and activate the account.
4. In the LMS, as `staff`:
    a. Add an EnterpriseCustomer and enable audit enrollment. Check its ID.
    b. Link the learner of `3` with the EnterpriseCustomer with no enrollment.
5. In Studio, as `staff`, add a new course. Check its ID.
6. In Ecommerce, as `staff`, add a Verified Course using the course ID of `5`. Use $500 for the price and include an Honor Seat.
7. In the LMS, as `staff`, go to the Course Modes in http://localhost:8000/admin/course_modes/coursemode/ and:
    a. Add an audit mode using the course ID of `5` with price 0
    b. Add a professional mode using the course ID of `5` with price 1000
8. Log out of the LMS as `staff` and log in as the learner linked in `4.b` and visit the landing page by typing an URL like this:

    http://localhost:8000/enterprise/2e347cae-7264-4652-ada9-f0d9e68021f9/course/course-v1:edX+MTC101+2017/enroll/

    where `2e347cae-7264-4652-ada9-f0d9e68021f9` should be the ID of the EnterpriseCustomer in `4.a` and `course-v1:edX+MTC101+2017` should be the ID of the course created in `5`.
9. Verify that the modes are displayed in the following order:
    * Verified
    * Professional
    * Audit
    * Honor
10. Optionally, you could modify the value of the `enterprise.constants.COURSE_MODE_SORT_ORDER` list and verify that the modes are displayed in the new order.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.~

**Post merge:**

**Reviewers:**
